src/plugins/intel_cpu/src/transformations/cpu_opset/common/op/vnode.cpp

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "vnode.hpp"
+
+#include <algorithm>
+
+#include "transformations/itt.hpp"
+
+ov::intel_cpu::VNode::VNode(const OutputVector& args, const Config& cfg) : Op(args), m_config(cfg) {
+    constructor_validate_and_infer_types();
+}
+
+std::shared_ptr<ov::Node> ov::intel_cpu::VNode::clone_with_new_inputs(
+    const ov::OutputVector& new_args) const {
+    INTERNAL_OP_SCOPE(VNode_with_new_inputs);
+    check_new_args_count(this, new_args);
+    return std::make_shared<ov::intel_cpu::VNode>(new_args, m_config);
+}
+
+void ov::intel_cpu::VNode::validate_and_infer_types() {
+    INTERNAL_OP_SCOPE(VNode_validate_and_infer_types);
+    if (m_config.type == "CausalMaskPreprocess") {
```

Comment:
Will there be other types ?

Reply:
I suppose in the future, more such big-pattern can share same ngraph VNode & CPU VNode